.github/workflows/app-build-and-deploy.yml

```diff
@@ -0,0 +1,124 @@
+        required: true
+
+    inputs:
+      deploy-env:
+        description: 'environment to deploy (i.e. dev | prod)'
+        required: true
+        type: string
+      application-type:
+        description: 'application type (i.e. api | worker | ui)'
+        required: true
+        type: string
+      azure-app-base-name:
+        description: 'Azure application name of webapp to deploy (i.e. clearlydefined-api | cdcrawler | clearlydefined)'
+        required: true
+        type: string
+      azure-app-name-postfix:
+        description: 'postfix to apply to the base name for the primary deploy site (e.g. -prod, -dev)'
+        required: true
+        type: string
+      secondary-azure-app-name-postfix:
+        description: 'postfix to apply to the base name for a secondary deploy site (e.g. -prod-europe, do not specify if no secondary site)'
+        type: string
+        default: ''
+      is-release:
+        description: 'true only when triggered by a release; otherwise false when dev deploy or manually deploy of prod'
+        type: boolean
+        default: false   
+
+jobs:
+  get-version:
+    name: Get version from package-lock.json
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ env.VERSION }}
+    steps:
+      - uses: actions/checkout@v4.1.1        
+      - name: Get version from package-lock.json
+        id: get_version
+        shell: bash
+        run: |
+          version='v'$(jq -r '.version' package-lock.json)       # e.g. v1.2.0
+          if [[ ${{ inputs.deploy-env }} == 'prod' ]]; then
+            if [[ ${{ inputs.is-release }} == 'true' ]]; then
+              # validate the version when triggered by a release
+              if [[ $version != 'v'${{ github.event.release.tag_name }} ]]; then
+                echo "Version in package-lock.json ($version) does not match the release tag (${{ github.event.release.tag_name }})"
+                exit 1
+              fi
+            fi
+          elif [[ ${{ inputs.deploy-env }} == 'dev' ]]; then
+            short_sha=$(echo "${{ github.sha }}" | cut -c 1-10)
+            version=$version'-dev-'$short_sha                    # e.g. v1.2.0-dev:1234567890
+          else
+            echo "Invalid deploy-env: ${{ inputs.deploy-env }}. Must be 'dev' or 'prod'"
+            exit 1
+          fi
+
+          echo "VERSION=$version" >> $GITHUB_ENV
+          echo "BuildAndDeploy: get-version -> outputs -> version: $version"
+
+  build-and-publish-image:
+    name: Build and publish Docker image
+    needs: get-version
+    uses: clearlydefined/operations/.github/workflows/app-build-docker-image.yml@elr/reusable-deploy-workflow
+    secrets:
+      DEPLOY_TOKEN: ${{ secrets.DEPLOY_TOKEN }}
+      PRODUCTION_DEPLOYERS: ${{ secrets.PRODUCTION_DEPLOYERS }}
+    with:
+      deploy-env: ${{ inputs.deploy-env }}
+      application-type: ${{ inputs.application-type }}
+      image-tag: ${{ needs.get-version.outputs.version }}
+
+  deploy-primary-app-to-azure:
```

Comment:
was thinking we could do a matrix thing here to dry up even more but may over complicate it. 🤔 

The only diffs between `deploy-primary-app-to-azure` and `deploy-secondary-app-to-azure` are:

- AZURE_WEBAPP_PUBLISH_PROFILE value
- azure-webapp-name

Reply:
Yes, I was thinking the same thing, but it didn't seem worth it unless we end up adding another one.  But I'm definitely open to a suggestion on how that would be done.  I agree it would be cleaner.

Reply:
* https://github.com/clearlydefined/operations/issues/74 captures information about potentially doing a refactor to use a matrix to identify deploy sites.